File: assets/samples/domain.availability/apikey/python/domain.avail.py
try:
    from urllib.request import urlopen
except ImportError:
    from urllib2 import urlopen

import base64
import hashlib
import hmac
import time

domain = 'example.com'
key = 'your domain availability api key'
secret = 'your domain availability api secret key'
username = 'your domain availability api username'

time = int(round(time.time() * 1000))
req = base64.b64encode(
    ('{"t":' + str(time) + ',"u":"' + username + '"}').encode('ascii')
)
digest = hmac.new(
    secret.encode('ascii'), 
    (username + str(time) + key).encode('ascii'), 
    hashlib.md5
).hexdigest()

url = 'http://www.whoisxmlapi.com/whoisserver/WhoisService?'\
    + 'requestObject=' + req.decode('ascii') + '&digest=' + digest \
    + '&domainName=' + domain + '&cmd=GET_DN_AVAILABILITY'

print(urlopen(url).read().decode('utf8'))

File: assets/samples/domain.availability/apikey/nodejs/domain.avail.js
var crypto = require('crypto');
var http = require('http');

var domain = 'example.com';
var key = 'Your domain availability api public key';
var secret = 'Your domain availability api secret key';
var username = 'Your domain availability api username';

var time = (new Date).getTime();
var req = Buffer.from('{"t":'+time+',"u":"'+username+'"}').toString('base64');
var hmac = crypto.createHmac('md5', secret).update(username + time + key);
var digest = hmac.digest('hex');

var url = 'http://www.whoisxmlapi.com/whoisserver/WhoisService?'
        + 'requestObject=' +req + '&digest=' +digest + '&domainName=' +domain
        + '&cmd=GET_DN_AVAILABILITY';

http.get(url, function(response) {
    var str = '';
    response.on('data', function(chunk) { str += chunk; });
    response.on('end', function() { console.log(str); });
}).end();